### PR TITLE
docs: update tsdocs to reduce gaps

### PR DESCRIPTION
**Pull request by jjarvisp**
#### Description of changes
Update TS Docs for the following:
- Analytics - disable
- Analytics - enable
- Analytics - flushEvents
- API - generateClient
- API - isCancelError
- Auth - decodeJWT
- Auth - fetchAuthSession
- In-app-messaging - initializeInAppMessaging
- Storage - isCancelError

#### Description of how you validated changes
Viewed generated changes on documentation site locally.

#### Checklist

- [X] PR description included
- [ ] `yarn test` passes
- [ ] Unit Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

#### Checklist for repo maintainers

- [ ] Verify E2E tests for existing workflows are working as expected or add E2E tests for newly added workflows
- [ ] New source file paths included in this PR have been added to CODEOWNERS, if appropriate

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/analytics/src/apis/disable.ts [hunk 1]
```diff
@@ -9,5 +9,6 @@ import { disableAnalytics } from '../utils';
  * @note
  * When Analytics is disabled events will not be buffered or transmitted to your selected service. Any auto-tracking
  * behavior that you have configured via `configureAutoTrack` will not have any effect while Analytics is disabled.
+ * @note Does not throw an exception.
```
Comment (jimblanc) [inline packages/analytics/src/apis/disable.ts:12]: Nit but personally I would leave these notes out here and elsewhere, I don't think they really add value

File: packages/core/src/singleton/apis/fetchAuthSession.ts [hunk 2]
```diff
@@ -6,6 +6,15 @@ import { AuthSession, FetchAuthSessionOptions } from '../Auth/types';
 
 import { fetchAuthSession as fetchAuthSessionInternal } from './internal/fetchAuthSession';
 
+/**
+ * Fetch the auth tokens, and the temporary AWS credentials and identity if they are configured. By default it
+ * does not refresh the auth tokens or credentials if they are loaded in storage already. You can force a refresh
+ * with `{ forceRefresh: true }` input.
```
Comment (jimblanc) [inline packages/core/src/singleton/apis/fetchAuthSession.ts:12]: nit

```suggestion
 * Fetch the auth session including the tokens and credentials if they are available. By default it
 * does not refresh the auth tokens or credentials if they are loaded in storage already. You can force a refresh
 * with `{ forceRefresh: true }` input.
```

**Review by jimblanc — COMMENTED** [review]
Looks good, few nits

File: packages/analytics/src/apis/disable.ts (hunk 1 above)
Comment (jjarvisp) [reply packages/analytics/src/apis/disable.ts]: Thanks Jim. I agree, I've removed these call outs

**Review by jimblanc — APPROVED** [review]

**Review by cshfang — APPROVED** [review]
Super nit: A bit inconsistent about having period vs not - but I don't think that's something we're consistent about in general throughout so it's a non-issue really

**Review by ashika112 — APPROVED** [review]

**Pull request merged**